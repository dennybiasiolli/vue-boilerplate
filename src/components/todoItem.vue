<template>
  <v-list-tile @click="()=>{}">
    <v-list-tile-action>
      <v-icon color="indigo">
        keyboard_arrow_right
      </v-icon>
    </v-list-tile-action>
    <v-list-tile-content>
      <v-list-tile-title>{{ todo.text }}</v-list-tile-title>
      <v-list-tile-sub-title>{{ todo.text }}</v-list-tile-sub-title>
    </v-list-tile-content>
  </v-list-tile>
</template>

<script>
export default {
  props: {
    todo: {
      type: Object,
      required: true
    }
  }
}
</script>
