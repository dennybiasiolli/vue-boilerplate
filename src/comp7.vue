<template>
  <div>
    <ol>
      <todo-item v-for="item in groceryList" v-bind:todo="item" v-bind:key="item.id">
      </todo-item>
    </ol>
  </div>
</template>

<script>
import todoItem from '@/todoItem'

export default {
  data() {
    return {
      groceryList: [
        { id: 0, text: 'Vegetables' },
        { id: 1, text: 'Cheese' },
        { id: 2, text: 'Whatever else humans are supposed to eat' }
      ]
    }
  },
  components: {
    'todo-item': todoItem
  }
}
</script>
