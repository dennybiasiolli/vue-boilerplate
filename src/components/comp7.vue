<template>
  <v-list>
    <todo-item
      v-for="item in groceryList"
      :key="item.id"
      :todo="item"
    />
  </v-list>
</template>

<script>
import todoItem from '@/components/todoItem'

export default {
  components: {
    'todo-item': todoItem
  },
  data() {
    return {
      groceryList: [
        { id: 0, text: 'Vegetables' },
        { id: 1, text: 'Cheese' },
        { id: 2, text: 'Whatever else humans are supposed to eat' }
      ]
    }
  }
}
</script>
