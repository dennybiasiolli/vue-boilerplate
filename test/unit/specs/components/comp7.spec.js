import { shallow, mount } from 'vue-test-utils'
import comp7 from '@/components/comp7'
import todoItem from '@/components/todoItem'

describe('comp7', () => {
  let component, vm, element

  beforeEach(() => {
    component = shallow(comp7)
    vm = component.vm
    element = component.element
  })

  test('should match snapshot', () => {
    expect(element).toMatchSnapshot()
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct default data', () => {
    expect(comp7.components).toEqual({
      'todo-item': todoItem
    })
    expect(comp7.components['todo-item']).toBe(todoItem)
    expect(typeof comp7.data).toBe('function')
    const defaultData = comp7.data()
    expect(defaultData.groceryList).toEqual([
      { id: 0, text: 'Vegetables' },
      { id: 1, text: 'Cheese' },
      { id: 2, text: 'Whatever else humans are supposed to eat' }
    ])
  })

  // Inspect the component instance on mount
  test('correctly sets the seen flag when created', () => {
    expect(vm.groceryList).toEqual([
      { id: 0, text: 'Vegetables' },
      { id: 1, text: 'Cheese' },
      { id: 2, text: 'Whatever else humans are supposed to eat' }
    ])
  })

  // Mount an instance and inspect the render output
  test('renders the correct message', () => {
    const element = mount(comp7).element
    expect(element.textContent).toContain('Vegetables')
    expect(element.textContent).toContain('Cheese')
    expect(element.textContent).toContain('Whatever else humans are supposed to eat')
  })
})
