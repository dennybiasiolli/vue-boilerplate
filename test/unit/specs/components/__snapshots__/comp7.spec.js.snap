// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comp7 should match snapshot 1`] = `
<v-list-stub>
  <todo-item-stub
    todo="[object Object]"
  />
  <todo-item-stub
    todo="[object Object]"
  />
  <todo-item-stub
    todo="[object Object]"
  />
</v-list-stub>
`;
